source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h

```diff
@@ -0,0 +1,124 @@
+  const Network::Address::InstanceConstSharedPtr& remoteAddress() const override;
+  const Network::Address::InstanceConstSharedPtr& localAddress() const override;
+  absl::optional<Network::Connection::UnixDomainSocketPeerCredentials>
+  unixSocketPeerCredentials() const override {
+    ASSERT(false, "Unix domain socket is not supported.");
+    return absl::nullopt;
+  }
+  void setConnectionStats(const Network::Connection::ConnectionStats& stats) override {
+    stats_ = std::make_unique<Network::Connection::ConnectionStats>(stats);
+    quic_connection_->setConnectionStats(stats);
+  }
+  Ssl::ConnectionInfoConstSharedPtr ssl() const override;
+  Network::Connection::State state() const override {
+    return quic_connection_->connected() ? Network::Connection::State::Open
+                                         : Network::Connection::State::Closed;
+  }
+  void write(Buffer::Instance& /*data*/, bool /*end_stream*/) override {
+    // All writes should be handled by Quic internally.
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+  void setBufferLimits(uint32_t limit) override;
+  uint32_t bufferLimit() const override;
+  bool localAddressRestored() const override {
+    // SO_ORIGINAL_DST not supported by QUIC.
+    return false;
+  }
+  bool aboveHighWatermark() const override {
+    ENVOY_CONN_LOG(error, "QUIC doesn't have connection level write buffer limit.", *this);
+    return false;
+  }
+  const Network::ConnectionSocket::OptionsSharedPtr& socketOptions() const override;
+  StreamInfo::StreamInfo& streamInfo() override { return stream_info_; }
+  const StreamInfo::StreamInfo& streamInfo() const override { return stream_info_; }
+  absl::string_view transportFailureReason() const override { return transport_failure_reason_; }
+
+  // Network::FilterManagerConnection
+  void rawWrite(Buffer::Instance& data, bool end_stream) override;
+
+  // Network::ReadBufferSource
+  Network::StreamBuffer getReadBuffer() override {
+    // Network filter has to stop iteration to prevent hitting this line.
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+  // Network::WriteBufferSource
+  Network::StreamBuffer getWriteBuffer() override { NOT_REACHED_GCOVR_EXCL_LINE; }
+
+protected:
+  // Propagate connection close to network_connection_callbacks_.
+  void onConnectionCloseEvent(const quic::QuicConnectionCloseFrame& frame,
+                              quic::ConnectionCloseSource source);
+
+  void raiseEvent(Network::ConnectionEvent event);
+
+  std::unique_ptr<EnvoyQuicConnection> quic_connection_;
+  // TODO(danzh): populate stats.
+  std::unique_ptr<Network::Connection::ConnectionStats> stats_;
+
+private:
+  // Currently ConnectionManagerImpl is the one and only filter. If more network
+  // filters are added, ConnectionManagerImpl should always be the last one.
+  // Its onRead() is only called once to trigger ReadFilter::onNewConnection()
+  // and the rest incoming data bypasses these filters.
+  Network::FilterManagerImpl filter_manager_;
```

Comment:
I think this is related to my other question. I'm wondering if we can just kill the filter manager entirely? Can you walk me through what we need it for?

Reply:
see my comments in the other place